### PR TITLE
Pelahi/mpi updates

### PR DESCRIPTION
**Pull request by pelahi**
Just updates to mpich container recipe (added metadata, added mpi4py since we need that too, added some more tests). Also added a recipe for building slurm in a container that works with Setonix and also spack (which is setup to build on top of the mpich image). The spack+mpi container is the critical one for groups that are really going to build a LOT of stuff in their container. Want to avoid the possibility of pulling in conflicting libraries (libcurl in particular) that cause issues when mounting libraries from Setonix into the container.

### CODE REVIEW

File: .github/workflows/docker-image.yml
Comment (marcodelapierre) [inline .github/workflows/docker-image.yml:1]: this is amazing, thanks Pascal!

**Review by marcodelapierre — CHANGES_REQUESTED** [review]
I think my only change suggestions are about the Spack dockerfile, see related comments.

I have left the Slurm one review to @ddeeptimahanti

File: spack/spack-mpich/buildspack.dockerfile
[the remark this answers is not in the document]
Comment (pelahi) [reply spack/spack-mpich/buildspack.dockerfile:1]: The main issue is then adding the mpi external for spack. But I'll figure out a clean way of doing this.

**Review by AlexisEspinosaGayosso — APPROVED** [review]
Seems fine to me.

**Review by AlexisEspinosaGayosso — APPROVED** [review]

**Review by AlexisEspinosaGayosso — APPROVED** [review]
All this looks fine for me. It would be nice to provide working examples on Setonix that we can access and test quickly rather than building the containers ourselves and think of ways for testing them.

File: spack/spack-mpich/buildspack.dockerfile
[the remark this answers is not in the document]
Comment (pelahi) [reply spack/spack-mpich/buildspack.dockerfile:1]: What I'll do is have two recipes, one that builds on the pawsey mpi and the other more generic.

**Pull request merged**